### DISCUSSION TITLE
Right Click terminal is not working in laragon 6.

### DISCUSSION THREAD

**Discussion by mh-shuvo** [leokhoa/laragon#545] [General]
Recently I installed laragon 6 in computer which is used windows 10. And I installed it in different drive. 
After installed I got the option in the right click like Terminal. But When I click on the Terminal it's throw an error like Application not found.  I couldn't understand that it's my installation problem or something else. I attached the screenshot here.
![Screenshot (2)](https://github.com/leokhoa/laragon/assets/30872044/1c751cd5-55bc-48e2-ad6a-6aa48bdde134)
![Screenshot (3)](https://github.com/leokhoa/laragon/assets/30872044/7109043c-737f-4db4-919c-bd300ae27e8c)

**Comment by dv007** [comment]
The error you mentioned I got, the cause is that the path to the terminal executable is wrong.
Here is the solution:

- Download the software **ShellMenuView** (http://www.nirsoft.net/utils/shell_menu_view.html)
- Run the software and find the line **Terminal**
- Right click on the **Terminal** line and select **Open In RegEdit**, you will see the path of the Terminal menu. There are 2 options to solve it: fix path to "laragon\bin\cmder\Cmder.exe" or create file cmder.bat (the contents of the file can be found here https://github.com/leokhoa/laragon/blob/6.0.0/bin/cmder/cmder.bat, but I chose option 1 because it's simpler!)
- Restart the computer

**Reply by canonprinter4** [reply]
I still don't understand the explanation above, can you explain in as much detail as possible? Please help me

**Comment by canonprinter4** [comment]
I still don't understand the explanation above, can you explain in as much detail as possible? Please help me

**Comment by ionut** [comment]
I can explain you in more details. 
1(first step) download from http://www.nirsoft.net/utils/shell_menu_view.html the software. scroll down till you see some blue anchor link.
![image](https://github.com/leokhoa/laragon/assets/24230102/b2e1879d-09e7-4298-adea-cbf012be2691)
2(second step) unzip de file that you downloaded previous step and start the shmnview application. in the application you will find some list with a lot of files. scroll to Terminal->right click->Open in RegEdit
![image](https://github.com/leokhoa/laragon/assets/24230102/16030e66-afa1-4e42-8800-36b2157f4803)
3(third step) right click on the string->Modify->change Value data with your location(find where do you install laragon, find folder bin, than folder cmder and you will see your cmder app). mine was by default "F:\laragon\bin\cmder\cmder.bat" "%v" and i changed it to "F:\laragon\bin\cmder\Cmder.exe" "%v". As you can see from the next print screen my file app was capitalize(Cmder) and  .exe not .bat.
![image](https://github.com/leokhoa/laragon/assets/24230102/92f645af-21bd-49ea-a5fd-2c0b462c35d5)

![image](https://github.com/leokhoa/laragon/assets/24230102/2025c1df-0bab-451f-ae1c-b4c09811095b)
4(fourth step) restart pc and all should be good.

**Comment by bilalMJHD** [comment]
If you are too lazy watch the gif image which i uploaded.
This is best I could do to fix this issue to open laragon terminal through right click context menu on windows.
Open Windows Run Menu with "Win + R" from keyboard and type "regedit" to open Registry Editor.
Now Press "Ctrl + F " to find the term "laragon-terminal".
After the registry folder show up expand the "00-laragon-terminal' folder and click on "command" folder.
By clicking on command folder you will see the key in the right side pane named "Default".
Double click on "Default" key and you will see the path of laragon terminal called "cmder"
Just rename the "cmder.bat" to "cmder.exe" and you are good to go.

![Laragon Terminal Error](https://github.com/leokhoa/laragon/assets/35788870/2d738d9e-08b5-4c84-9060-09bbf34c5f7c)

**Comment by anis14109** [comment]
Here is a one-click solution with a batch (.bat) script.
Just create a file `fix-terminal.bat`, copy-paste the batch code &
run it iwith administrative privileges, Done.

```
@echo off
:: Ensure the script is running with administrative privileges
net session >nul 2>&1
if %errorLevel% neq 0 (
    echo.
    echo ERROR: This script must be run as an Administrator!
    echo Please right-click the file and select 'Run as administrator'.
    echo.
    pause
    exit /b
)

echo Updating Laragon 00-laragon-terminal Registry keys...

:: Update the background shell command to cmder.exe
reg add "HKLM\SOFTWARE\Classes\Directory\background\shell\00-laragon-terminal\command" /ve /t REG_SZ /d "\"C:\laragon\bin\cmder\cmder.exe\" \"%%v\"" /f >nul 2>&1

:: Update the standard directory shell command to cmder.exe just in case
reg add "HKLM\SOFTWARE\Classes\Directory\shell\00-laragon-terminal\command" /ve /t REG_SZ /d "\"C:\laragon\bin\cmder\cmder.exe\" \"%%1\"" /f >nul 2>&1

if %errorLevel% equ 0 (
    echo.
    echo SUCCESS: Laragon Terminal mapped to cmder.exe successfully!
) else (
    echo.
    echo ERROR: Failed to update registry. Check your admin rights.
)

echo.
pause
```
